src/controllers/mainController.ts

```diff
@@ -385,87 +385,126 @@ export default class MainController implements vscode.Disposable {
+                    scriptType = "Delete";
+                    break;
+                case ScriptOperation.Execute:
+                    scriptType = "Execute";
+                    break;
+                case ScriptOperation.Alter:
+                    scriptType = "Alter";
+                    break;
+                default:
+                    scriptType = "Unknown";
+                    break;
+            }
+            sendActionEvent(
+                TelemetryViews.QueryEditor,
+                TelemetryActions.RunQuery,
+                {
+                    isScriptExecuted: executeScript.toString(),
+                    objectType: node.nodeType,
+                    operation: scriptType,
+                },
                 undefined,
-                title,
-                {},
-                queryPromise,
-            );
-            await queryPromise;
-            await this.connectionManager.connectionStore.removeRecentlyUsed(
-                <IConnectionProfile>connectionCreds,
+                connectionCreds as IConnectionProfile,
+                this.connectionManager.getServerInfo(connectionCreds),
             );
-        }
+        };
 
-        let scriptType;
+        let operationType = "";
         switch (operation) {
             case ScriptOperation.Select:
-                scriptType = "Select";
+                operationType = LocalizedConstants.ObjectExplorer.ScriptSelectLabel;
                 break;
             case ScriptOperation.Create:
-                scriptType = "Create";
+                operationType = LocalizedConstants.ObjectExplorer.ScriptCreateLabel;
                 break;
             case ScriptOperation.Insert:
-                scriptType = "Insert";
+                operationType = LocalizedConstants.ObjectExplorer.ScriptInsertLabel;
                 break;
             case ScriptOperation.Update:
-                scriptType = "Update";
+                operationType = LocalizedConstants.ObjectExplorer.ScriptUpdateLabel;
                 break;
             case ScriptOperation.Delete:
-                scriptType = "Delete";
+                operationType = LocalizedConstants.ObjectExplorer.ScriptDeleteLabel;
                 break;
             case ScriptOperation.Execute:
-                scriptType = "Execute";
+                operationType = LocalizedConstants.ObjectExplorer.ScriptExecuteLabel;
                 break;
             case ScriptOperation.Alter:
-                scriptType = "Alter";
+                operationType = LocalizedConstants.ObjectExplorer.ScriptAlterLabel;
                 break;
             default:
-                scriptType = "Unknown";
-                break;
+                operationType = LocalizedConstants.ObjectExplorer.ScriptSelectLabel;
         }
-        sendActionEvent(
-            TelemetryViews.QueryEditor,
-            TelemetryActions.RunQuery,
+
+        await vscode.window.withProgress(
```

Comment:
Another progress toast?  When does this one show up?  Maybe I missed it in your videos.